pkg/auth/githubapp.go

```diff
@@ -0,0 +1,78 @@
+// Copyright 2023 The Authors (see AUTHORS file)
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package auth
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/abcxyz/pkg/githubapp"
+)
+
+// ReadAccessTokenForRepos generate a new access token with read permissions
+// for the given repositories using the given GitHub App.
+func ReadAccessTokenForRepos(ctx context.Context, githubApp *githubapp.GitHubApp, repositories ...string) (string, error) {
+	tokenRequest := &githubapp.TokenRequest{
+		Repositories: repositories,
+		Permissions: map[string]string{
+			"actions": "read",
+		},
+	}
+	// @TODO(bradegler): This could use some caching. Requests to the same repos
+	// can reuse a token without requesting a new one until it expires. Might be
+	// better to implement that in pkg so that GitHubTokenMinter can take
+	// advantage of it as well.
+	ghAppJWT, err := githubApp.AccessToken(ctx, tokenRequest)
+	if err != nil {
+		return "", fmt.Errorf("error creating GitHub access token: %w", err)
+	}
+	return parseJWT(ghAppJWT)
+}
+
+// ReadAccessTokenForAllRepos generate a new access token with read permissions
```

Comment:
Isn't this duplicative with functionality in another repo already?

Reply:
I'm not aware of any other repo that has a similar function, was there one you had in mind?

Reply:
I could've sworn I saw a PR from @verbanicm that had this. Maybe it was the implementation for the client.